cloud/azure/bin/lib/key_vault.sh

```diff
@@ -33,15 +23,23 @@ function azure::create_vault() {
 
 #######################################
 # Assign the role 'Key Vault Secrets officer' to the current user
+# Arguments:
+#   1. The resource group to scope the role assignment to
 #######################################
-function azure::assign_secrets_officer_role_to_user() {
-  local user_id=$(az ad signed-in-user show --query objectId -o tsv)
-  local subscription_id=$(az account show --query id -o tsv)
-  az role assignment create \
-    --role "b86a8fe4-44ce-4948-aee5-eccb2c155cd7" \
-    --scope "subscriptions/${subscription_id}/resourceGroups/${resource_group}" \
-    --assignee-object-id "${user_id}" \
-    --assignee-principal-type "User"
+function key_vault::assign_secrets_officer_role_to_user() {
+  local USER_ID="$(az ad signed-in-user show --query objectId -o tsv)"
+  local SUBSCRIPTION_ID="$(az account show --query id -o tsv)"
+  local ROLE_ASSIGNMENTS="$(az role assignment list --assignee ${USER_ID})"
+  if [[ ("Key Vault Secrets Officer" == *$ROLE_ASSIGNMENTS*) ]];
```

Comment:
Use "${}" around ROLE_ASSIGNMENTS

I _think_ the only time you don't need "" is in a (( )) but I'm happy to be wrong.

https://google.github.io/styleguide/shellguide.html#variable-expansion